### PR TITLE
Equation of State workflow for FHI-aims

### PR DESCRIPTION
**Pull request by ansobolev**
## Summary

 * This PR adapts the common Equation of State workflow to FHI-aims. 

## Additional dependencies introduced (if any)

None

## TODO (if any)

None

## Checklist

Before a pull request can be merged, the following items must be checked:

* [x] Code is in the [standard Python style](https://www.python.org/dev/peps/pep-0008/).
  The easiest way to handle this is to run the following in the **correct sequence** on
  your local machine. Start with running [`ruff`](https://docs.astral.sh/ruff) and `ruff format` on your new code. This will
  automatically reformat your code to PEP8 conventions and fix many linting issues.
* [x] Doc strings have been added in the [Numpy docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html).
  Run [ruff](https://beta.ruff.rs/docs/rules/#pydocstyle-d) on your code.
* [x] Type annotations are **highly** encouraged. Run [mypy](http://mypy-lang.org) to
  type check your code.
* [x] Tests have been added for any new functionality or bug fixes.
* [x] All linting and tests pass.

Note that the CI system will run all the above checks. But it will be much more
efficient if you already fix most errors prior to submitting the PR. It is highly
recommended that you use the pre-commit hook provided in the repository. Simply run
`pre-commit install` and a check will be run prior to allowing commits.

### CODE REVIEW

File: src/atomate2/aims/flows/eos.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        default_factory=lambda: DoubleRelaxMaker.from_parameters({})
+    )
+    eos_relax_maker: Maker | None = field(
+        default_factory=lambda: RelaxMaker.fixed_cell_relaxation(
+            user_params={"species_dir": "tight"}
+        )
+    )
+
+    @classmethod
+    def from_parameters(cls, parameters: dict[str, Any], **kwargs) -> AimsEosMaker:
```
Comment (janosh) [inline src/atomate2/aims/flows/eos.py:58]: minor nitpick but i would prob call this `from_params` for brevity.

**Review by janosh — APPROVED** [review]
nice work, esp. on the tests! 👍 lgtm

File: src/atomate2/aims/flows/eos.py (hunk 1 above)
Comment (ansobolev) [reply src/atomate2/aims/flows/eos.py:58]: This method is usually called `from_parameters` in aims workflows, I'd leave it as it is for compatibility

**Comment by ansobolev** [comment]
Hi, is there anything else we can do with the pull request?

**Comment by janosh** [comment]
thanks again @ansobolev, i'll go ahead and merge this! 👍

**Pull request merged**